### DISCUSSION TITLE
token.getAddress is not a function

### DISCUSSION THREAD

**Discussion by 6be709c0** [NomicFoundation/hardhat#4004] [General]
Hello,

I'd be happy to do pull requests but I want to make sure that's actual deprecated code first.

Following `7. Deploying to a live network`

The code:
```
  console.log("Token address:", await token.getAddress());
 ```
  
  Returns `token.getAddress is not a function`
  Replacing it with: 
```
    console.log("Token address:", token.address);
```
works.

**Comment by fvictorio** [comment]
Hi @lescaudr, we just updated many of our plugins to use ethers v6 instead of v5. I think you were caught in the middle of that update, and probably started the tutorial by installing the old versions that are no longer compatible with the current content. Sorry about that!

You can read about these changes [here](https://github.com/NomicFoundation/hardhat/releases/tag/%40nomicfoundation%2Fhardhat-toolbox%403.0.0).
